deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/LayerConfiguration.java

```diff
@@ -0,0 +1,313 @@
+            regularizerFields.remove("l1");
+            regularizerFields.remove("l2");
+            regularizerFields.remove("name");
+            if (regularizerFields.size() > 0) {
+                String unknownField = (String) regularizerFields.toArray()[0];
+                log.warn("Unknown regularization field: " + unknownField);
+            }
+        }
+    }
+
+    /**
+     * Perform layer configuration steps that are common across all Keras and DL4J layer types.
+     *
+     * @param builder       DL4J Layer builder object
+     * @param kerasConfig   Map containing Keras layer properties
+     * @return              DL4J Layer builder object
+     * @throws NotImplementedException
+     * @see Layer.Builder
+     */
+    public static Layer.Builder finishLayerConfig(Layer.Builder builder, Map<String,Object> kerasConfig)
+        throws NotImplementedException {
+        if (kerasConfig.containsKey("dropout"))
+            builder.dropOut((double)kerasConfig.get("dropout"));
+        if (kerasConfig.containsKey("activation"))
+            builder.activation(mapActivation((String)kerasConfig.get("activation")));
+        builder.name((String)kerasConfig.get("name"));
+        if (kerasConfig.containsKey("init")) {
+            WeightInit init = mapWeightInitialization((String) kerasConfig.get("init"));
+            builder.weightInit(init);
+            if (init == WeightInit.ZERO)
+                builder.biasInit(0.0);
+        }
+        if (kerasConfig.containsKey("W_regularizer")) {
+            Map<String,Object> regularizerConfig = (Map<String,Object>)kerasConfig.get("W_regularizer");
+            double l1 = getL1Regularization(regularizerConfig);
+            if (l1 > 0)
+                builder.l1(l1);
+            double l2 = getL2Regularization(regularizerConfig);
+            if (l2 > 0)
+                builder.l2(l2);
+            checkForUnknownRegularizer(regularizerConfig);
+        }
+        if (kerasConfig.containsKey("b_regularizer")) {
+            Map<String,Object> regularizerConfig = (Map<String,Object>)kerasConfig.get("b_regularizer");
+            double l1 = getL1Regularization(regularizerConfig);
+            double l2 = getL2Regularization(regularizerConfig);
+            if (l1 > 0 || l2 > 0)
+                throw new NotImplementedException("Bias regularization not implemented");
+        }
+        return builder;
+    }
+
+    /**
+     * Configure DL4J DenseLayer from a Keras Dense configuration.
+     *
+     * @param kerasConfig      Map containing Keras Dense layer properties
+     * @return                 DL4J DenseLayer configuration
+     * @throws NotImplementedException
+     * @see DenseLayer
+     */
+    public static DenseLayer buildDenseLayer(Map<String,Object> kerasConfig)
+        throws NotImplementedException {
+        DenseLayer.Builder builder = new DenseLayer.Builder()
+                .nOut((int)kerasConfig.get("output_dim"));
```

Comment:
These "magic strings" should be constants with public final static SOME_VARIABLE_NAME.